sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -847,48 +847,53 @@
      */
 
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PollerFlux<Void, Void> beginPurchaseSearch(String searchId, Duration pollInterval) {
+    public PollerFlux<PhoneNumberSearch, 
+        PhoneNumberSearch> beginPurchaseSearch(String searchId, Duration pollInterval) {
         Objects.requireNonNull(searchId, "'searchId' can not be null.");
-        Objects.requireNonNull(pollInterval, "'pollInterval' can not be null.");
+        
+        if (pollInterval == null) {
+            pollInterval = Duration.ofSeconds(5);
+        }
 
-        return new PollerFlux<Void, Void>(pollInterval,
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(pollInterval,
             purchaseSearchActivationOperation(searchId),
             purchaseSearchPollOperation(searchId),
             (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
             purchaseSearchFetchResultOperation());
     }
 
-    private Function<PollingContext<Void>, Mono<Void>> purchaseSearchActivationOperation(String searchId) {
-
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> purchaseSearchActivationOperation(String searchId) {
         return (pollingContext) -> {
-            Mono<Void> response = purchaseSearch(searchId);
-            return response;
+            Mono<PhoneNumberSearch> res = purchaseSearch(searchId).flatMap(agh -> {
```

Comment:
Can we use more descriptive parameter names than res, agh?

Reply:
Also, I find it weird that the argument in flatmap is not used.. is that normal pattern?

Reply:
I used agh to test, forgot to remove that!